src/properties/class-papi-property-term.php

```diff
@@ -14,85 +14,231 @@ class Papi_Property_Term extends Papi_Property {
+			],
+			'layout'        => 'single', // Single or multiple
+			'placeholder'   => '',
+			'taxonomy'      => '',
+			'select2'       => true,
+			'query'         => []
 		];
 	}
 
 	/**
-	 * Get terms.
+	 * Get labels from taxonomies.
 	 *
-	 * @param  stdClass $settings
+	 * @return array
+	 */
+	protected function get_labels() {
+		$results = [];
+
+		foreach ( $this->get_taxonomies() as $taxonomy ) {
+			if ( taxonomy_exists( $taxonomy ) ) {
+				$taxonomy_object    = get_taxonomy( $taxonomy );
+				$results[$taxonomy] = $taxonomy_object->labels->name;
+			}
+		}
+
+		return $results;
+	}
+
+	/**
+	 * Get post types.
 	 *
 	 * @return array
 	 */
-	protected function get_terms( $settings ) {
+	protected function get_taxonomies() {
+		return papi_to_array( $this->get_setting( 'taxonomy' ) );
+	}
+
+	/**
+	 * Get terms for specified taxonomy.
+	 * @param  string $taxonomy
+	 * @return array
+	 */
+	protected function get_terms( $taxonomy ) {
 		// Prepare arguments for get_terms.
-		$args = array_merge( $settings->query, [
+		$query  = $this->get_setting( 'query' );
+		$args = array_merge( $query, [
 			'fields' => 'id=>name'
 		] );
 
-		return get_terms( $settings->taxonomy, $args );
+		$terms = [];
+		if ( taxonomy_exists( $taxonomy ) ) {
+			$terms = get_terms( $taxonomy, $args );
+		}
+
+		 return $terms;
 	}
 
 	/**
-	 * Render property html.
+	 * Get single term.
+	 * @param  int $term_id
+	 * @return object
 	 */
-	public function html() {
-		$settings   = $this->get_settings();
-		$value      = $this->get_value();
-
-		if ( is_object( $value ) ) {
-			$value = $value->term_id;
-		} else {
-			$value = 0;
+	protected function get_term( $term_id ) {
+		if ( version_compare( $GLOBALS['wp_version'], '4.4', '<' ) ) {
+			$taxonomies = $this->get_taxonomies();
+			$taxonomy = reset( $taxonomies );
+			$term = get_term( $term_id, $taxonomy );
+		}
+		else {
+			$term = get_term( $term_id );
 		}
 
-		$classes = 'papi-fullwidth';
+		return $term;
+	}
+
+	/**
+	 * Render property html.
+	 */
+	public function html() {
+		$settings                = $this->get_settings();
+		$layout                  = $settings->layout;
+		$labels                  = $this->get_labels();
+		$taxonomies              = $this->get_taxonomies();
+		$render_label            = count( $taxonomies ) > 1;
+		$multiple                = $render_label && $layout === 'multiple';
+		$single                  = $render_label && $layout !== 'multiple';
+		$classes                 = count( $taxonomies ) > 1 ? '' : 'papi-fullwidth';
+		$value                   = $this->get_value();
+		$value                   = is_object( $value ) ? $value->term_id : 0;
+		$selected_label          = reset( $labels );
+		$selected_term           = $this->get_term( $value ) ? : '';
+		$selected_term           = is_wp_error( $selected_term ) || empty( $selected_term ) ? '' : $selected_term;
+		$selected_taxonomy       = empty( $selected_term ) ? reset( $taxonomies ) : $selected_term->taxonomy;
```

Comment:
Maybe remove some spacing since the aren't a variable that is so long, the longest variable should diced the spacing :)